### PR TITLE
tools: Update md-magic template infra, and update contribution guidelines template to match root README

### PR DESCRIPTION
**Pull request by Josmithr**
Updates our markdown-magic logic in the following ways:
- Simplifies code with new helper functionality
- Extracts basic templates into md files
- Ensures the generated content disclaimer is included in all gen'd doc sections

Also updates the contribution guidelines template to match the content in the repo README, and updates that README to use md-magic.

The primary changes to review are:
- `./docs/md-magic-config.js`
- the new md files under `./md-magic-templates`

The remaining changes are all generated using md-magic.

### CODE REVIEW

File: docs/md-magic.config.js [hunk 1]
```diff
@@ -92,6 +91,36 @@ const toPosix = (path) => {
     return inPosix;
 }
 
+/**
+ * Reads and returns the contents for the specified template file.
```
Comment (Josmithr) [inline docs/md-magic.config.js:95]:
```suggestion
 * Reads and returns the contents from the specified template file.
```

File: docs/md-magic.config.js [hunk 2]
```diff
@@ -92,6 +91,36 @@ const toPosix = (path) => {
...
+ *
+ * @param {string} templateFileName - Name of the file to read, under {@link mdMagicTemplatesPath}.
```
Comment (Josmithr) [inline docs/md-magic.config.js:97]:
```suggestion
 * @param {string} templateFileName - Name of the file to read, under {@link mdMagicTemplatesPath} (e.g. "Trademark-Template.md").
```

File: docs/content/docs/data-structures/string.md [hunk 3]
```diff
@@ -75,8 +79,6 @@ efficient spatial querying of the nearest Marker to a given position. -->
   - [collaborativeInput](https://github.com/microsoft/FluidFramework/blob/main/experimental/framework/react-inputs/src/CollaborativeInput.tsx)
 
 [SharedMap]: https://fluidframework.com/docs/data-structures/map/
-[SharedString]: https://github.com/microsoft/FluidFramework/blob/main/packages/dds/sequence/src/sharedString.ts
```
Comment (alexvy86) [inline docs/content/docs/data-structures/string.md:78]: I think the INCLUDE above on line 64 needs to be updated to include line 448 from the Sequence README.md:

image

**Review by alexvy86 — COMMENTED** [review]
LGTM

File: docs/content/docs/data-structures/string.md (hunk 3 above)
Comment (Josmithr) [reply docs/content/docs/data-structures/string.md:78]: Good catch! I completely missed this. Will update.

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**